### DISCUSSION TITLE
Esp32 vs Esp8266

### DISCUSSION THREAD

**Discussion by Xbox-gamer-tech** [luc-github/ESP3D#666] [Questions]
Is there any benefits/differences on using Esp32 instead of a Esp8266?

**Comment by luc-github** [comment]
* ESP32 is the successor of ESP8266 and will be replaced by ESP32 mono core version soon or later, more ESP32 versions are ongoing and will come.
* ESP32 has more memory and pins which allows more possible features
* ESP32 has 2 cores which allow to do something when wifi is also working when it is not possible with ESP8266
* Currently ESP32 is slower than in file transfer than ESP8266, this is due to ESP8266 is really really optimized, but I expect this gap to be cleared soon

I currently mostly focus on ESP32 but keep ESP8266/ESP8285 support because it still has a lot of users

**Reply by Xbox-gamer-tech** [reply]
Thanks for the response.

So yesterday I was finally able to get my Ender3 (ch340) to work with Max3421 (Esp3d was running on esp32 and the max3421 was not connected to the esp32 but a separate Arduino which was basically converting the data for ender . the only problem I faced was that I was not able to upload any files to the SD card.  ) regarding which I have few questions .

1)If I download the library and compile it myself dose the code work fine while uploading files to 3d printers cause mine kept starting random prints while uploading it could be also because I was using the usb host module with cheap jumper wires which could be causing signal problem while sending huge sets of data.

2) Which two variable do you store the data of sending and receiving commands from the printer.  

3)Dose anything run on the second core on the esp32 except the optional async web server.

4)Also if you have done any pogroming for the max3421 could I get that part of the code .


So right now what I am trying to do is to connected the amx3421 directly to the esp32 which is running esp3d and completely remove the extra Arduino in between which may also help stopping the issues which very caused while transferring files.

**Reply by luc-github** [reply]
1 - the serial file transfert is very slow and can be easily corrupted if FW do some auto send like temperature or anything, this is why I plan to work on https://github.com/luc-github/ESP3D/issues/575
2 - ESP3D FW is just a bridge, it directly transfert stream / commands from http / or tcp to serial , and from serial to websocket.
3 - Do not use the asyncweb server in 2.1 it is no more supported - it randomly loose data which make communication unreliable, core 0 is used by wifi, core 1 is used by ESP3D
4 - Yes did some test with https://github.com/felis/USB_Host_Shield_2.0 and got lot of devices not supported that need tuning, also it seems it use a lot of from ESP32 by some feedback, so I gave up (https://github.com/luc-github/ESP3D/issues/239) also I got another suggestion https://github.com/luc-github/ESP3D/issues/424 which seems more easy to implement but use private code, additionnaly some new esp32 come with some USB Host so ... https://github.com/luc-github/ESP3D/issues/550

**Reply by Xbox-gamer-tech** [reply]
So anyways if someone wants the code for using their printer which has ch340 and are ready to buy one more Arduino (which are supported by usb host 2.0) can just wait few more days and I will upload the code for it (uploading file from esp3d could be even more slow.)

**Reply by luc-github** [reply]
why do you think use usb host will speed up file transfert ? the speed issue is not related to usb or serial

**Reply by Xbox-gamer-tech** [reply]
No what I mean is if you use the usb host it will get slower.

**Reply by luc-github** [reply]
Ho - I miss read - sorry , well I am not sure, the speed issue is due to M29/M28 protocole itself so I guess it will be barely noticeable
good luck in your task
